### PR TITLE
ath79-generic: add support for TP-Link CPE220 v3

### PR DESCRIPTION
**Pull request by SunboX**
- [x] must be flashable from vendor firmware
  - [x] webinterface
  - [x] tftp
  - [ ] other: <specify>
- [x] must support upgrade mechanism
  - [x] must have working sysupgrade
    - [x] must keep/forget configuration (if applicable)
      *think `sysupgrade [-n]` or `firstboot`*
  - [x] must have working autoupdate
    *usually means: gluon profile name must match image name*
- [x] reset/wps/phone button must return device into config mode
- [x] primary mac should match address on device label (or packaging) (https://gluon.readthedocs.io/en/latest/dev/hardware.html#notes)
- wired network
  - [x] should support all network ports on the device
  - [x] must have correct port assignment (WAN/LAN)
- wifi (if applicable)
  - [x] association with AP must be possible on all radios
  - [x] association with 802.11s mesh must be working on all radios 
  - [x] ap/mesh mode must work in parallel on all radios
- led mapping
  - power/sys led (_critical, because led definitions are setup on firstboot only_)
    - [x] lit while the device is on
    - [x] should display config mode blink sequence 
(https://gluon.readthedocs.io/en/latest/features/configmode.html)
  - radio leds
    - [x] should map to their respective radio
    - [x] should show activity
  - switchport leds
    - [x] should map to their respective port (or switch, if only one led present) 
    - [x] should show link state and activity
- outdoor devices only
  - [ ] added board name to `is_outdoor_device` function in `package/gluon-core/luasrc/usr/lib/lua/gluon/platform.lua`

### CODE REVIEW

File: docs/user/supported_devices.rst [hunk 1]
```diff
@@ -79,7 +79,7 @@ ar71xx-generic
   - Archer C59 (v1)
   - Archer C7 (v2, v4, v5)
   - CPE210 (v1.0, v1.1, v2.0, v3.0)
-  - CPE220 (v1.1)
+  - CPE220 (v1.1, v3.0)
```
Comment (mweinelt) [inline docs/user/supported_devices.rst:82]: Would need to be put into the ath79-generic target, not ar71xx-generic. However I'm believe we need to flag this device as BROKEN, seeing as it uses the ath10k driver and only features 64 MB of RAM. 

CC @ecsv

**Comment by nalxnet** [comment]
Device is not supported in OpenWrt 19.07, only in master, so your PR needs to be rebased against next.

**Comment by ecsv** [comment]
> Device is not supported in OpenWrt 19.07, only in OpenWrt master, so your PR needs to be rebased against next.

No, it is supported. See the  0b6b2be4feca ("modules: bump OpenWrt") in master

    modules: bump OpenWrt
    
    e18c87ef45 ramips: fix wps leds/btn for TP-Link TL-WA801ND v5
    455ba76bf9 hostapd: cleanup IBSS-RSN
    dd4d49dcc1 cryptodev-linux: remove DEFAULT redefinition
    44b37774f9 mac80211: fix a page refcounting issue leading to leaks/crashes in rx A-MSDU decap
    a3b6ffe01b mac80211: fix sta TID stats leak on a few nl80211 calls
    25e1afb9e1 ucert: update to version 2019-12-19
    fe197b8b09 ramips: mt7621: disable images for gehua_ghl-r-001
    08d9828b76 ramips: fix leds for TP-Link Archer C20 v4
    fd28ef59db ath79: add SUPPORTED_DEVICES for TP-Link TL-WR841N/ND v9 to v12
    7a0d9b2eea ath79: add support for TP-Link TL-WR841N/ND v12
    bd3eb071fd ath79: add support for TP-Link TL-WR841N/ND v10
    19ff3f5105 ath79: add support for the TP-LINK CPE220 V3
    44c827215d ethtool: fix PKG_CONFIG_DEPENDS
    eb15634541 OpenWrt v19.07.0: revert to branch defaults
    aca39acedf OpenWrt v19.07.0: adjust config defaults
    a3ffeb413b ramips: Fix sysupgrade for Xiaomi mir3g
    f58705b77e dnsmasq: Fix potential dnsmasq crash with TCP
    54711e528d x86: fix missing led variable warning during boot
    abb0665bec ca-certificates: provide ca-certs by both ca-certificates and ca-bundle
    e9929ebeea ramips: Fix sysupgrade for Xiaomi mir3g
    dc399c4e12 ramips: remove duplicate dts nodes of MediaTek LinkIt Smart 7688
    a5653ec87e package: remove accidentally added symlink
    6395ac4126 fstools: update to latest Git HEAD
    f3439c4019 procd: update to version 2020-01-04
    64c45d95d6 ubus: update to version 2019-12-27
    04fd5e22b2 libubox: update to version 2019-12-28
    bf99f79200 base-files: sysupgrade: exit if the firmware download failed
    3140d38042 base-files: upgrade: add case to export_bootdevice
    3c11032039 sunxi: Turn on CONFIG_PINCTRL_SUN4I_A10 for A20
    3fc47dd443 wolfssl: bump to 4.3.0-stable
    330046922b kernel: bump 4.14 to 4.14.162
    084dfb8ebd kernel: bump 4.14 to 4.14.161
    
    Compile-tested: ath79-generic
    Run-tested: ath79-generic

> However I'm believe we need to flag this device as BROKEN, seeing as it uses the ath10k driver and only features 64 MB of RAM.

No, it doesn't even have ath10k installed:

    root@ffv-cc32e52d46bc:~# lsmod|grep ath
    ath                    19539  3 ath9k,ath9k_common,ath9k_hw
    ath9k                 117872  0 
    ath9k_common           12717  1 ath9k
    ath9k_hw              374474  2 ath9k,ath9k_common
    cfg80211              233940  5 ath9k,ath9k_common,batman_adv,ath,mac80211
    compat                  6519  5 ath9k,ath9k_common,batman_adv,mac80211,cfg80211
    mac80211              465192  2 ath9k,ath9k_hw

File: docs/user/supported_devices.rst (hunk 1 above)
Comment (ecsv) [reply docs/user/supported_devices.rst]: Section fixed

**Comment by nalxnet** [comment]
> > Device is not supported in OpenWrt 19.07, only in OpenWrt master, so your PR needs to be rebased against next.
> 
> No, it is supported. See the [0b6b2be](https://github.com/freifunk-gluon/gluon/commit/0b6b2be4feca17e1dcce459f0499de79163e2c74) ("modules: bump OpenWrt") in master

You're right, I looked at the wrong place, sorry about that. Realised it shortly after my comment and deleted it.

**Comment by mweinelt** [comment]
> No, it doesn't even have ath10k installed:
> 
> ```
> root@ffv-cc32e52d46bc:~# lsmod|grep ath
> ath                    19539  3 ath9k,ath9k_common,ath9k_hw
> ath9k                 117872  0 
> ath9k_common           12717  1 ath9k
> ath9k_hw              374474  2 ath9k,ath9k_common
> cfg80211              233940  5 ath9k,ath9k_common,batman_adv,ath,mac80211
> compat                  6519  5 ath9k,ath9k_common,batman_adv,mac80211,cfg80211
> mac80211              465192  2 ath9k,ath9k_hw
> ```

Ah I accidentally looked up the CPE210 v3, and that shows ath10k, which is weird.  https://openwrt.org/toh/hwdata/tp-link/tp-link_cpe210_v3

Any idea whats the relationship between CPE210 and CPE220?

**Comment by ecsv** [comment]
> Any idea whats the relationship between CPE210 and CPE220?

I don't have any official info but I think it was just about the ethernet. CPE210 has one and CPE220 has two.

**Comment by mweinelt** [comment]
LGTM. Thanks everyone!

**Pull request merged**